assets/src/edit-story/app/media/media3p/reducer.js

```diff
@@ -74,6 +75,12 @@ function reducer(state = INITIAL_STATE, { type, payload }) {
         selectedProvider: payload.provider,
       };
     }
+    case types.SEARCH: {
+      return {
+        ...state,
+        searchTerm: payload.searchTerm,
```

Comment:
This can also set 'pageToken: undefined' if the searchTerm has actually changed, so that fetching starts from the 1st page.